src/components/GrayFilter.vue

```diff
@@ -0,0 +1,43 @@
+<script lang="ts">
+import { defineComponent } from "vue";
+
+type Props = {
+  onClick: Function;
+};
+
+export default defineComponent({
+  name: "GrayFilter",
+  props: {
+    onClick: {
+      type: Function,
+      required: true,
+    },
+  },
+  emits: ["click"],
+  setup: (props: Props, { emit }) => {
+    const handleClick = (e: any) => {
+      emit("click", e.target.value);
+    };
```

Comment:
```suggestion
    const handleClick = () => {
      emit("click");
    };
```

clickはこれに統一しときます。
templateで`$emit("click")`だけでもいいかも